ui/Screen/Project/Commit.svelte

```diff
@@ -0,0 +1,131 @@
+<script>
+  import { Title, Flex, Code } from "../../DesignSystem/Primitive";
+  import { Icon } from "../../DesignSystem/Primitive";
+  import { gql } from "apollo-boost";
+  import { link } from "svelte-spa-router";
+  import { showNotification } from "../../store/notification.js";
+  import { getClient, query } from "svelte-apollo";
+  import { format } from "timeago.js";
+  import * as path from "../../lib/path.js";
+
+  export let params = null;
+  const projectId = params.id;
+  const commitHash = params.hash;
+
+  const QUERY = gql`
+    query($projectId: ID!, $commitHash: String!) {
+      commit(id: $projectId, sha1: $commitHash) {
+        author {
+          avatar
+          email
+          name
+        }
+        committerTime
+        message
+        sha1
+        summary
+      }
+    }
+  `;
+
+  async function fetchCommit() {
+    try {
+      const response = await query(getClient(), {
+        query: QUERY,
+        variables: {
+          projectId: projectId,
+          commitHash: commitHash
+        }
+      });
+      const result = await response.result();
+      const commit = result.data.commit;
+
+      // TODO(cloudhead): Fetch branch from backend.
+      commit.branch = "master";
```

Comment:
We need to provide this somehow from the proxy, when fetching a commit.

Reply:
What are the semantics of branch association for commits? How do we deal with it occurring in multiple? Is this specced already? If so we can then evaluate if we have the building blocks to implement it or need more functionality in surf.

Reply:
Would also reckon that in this view it's important to not only show the branch but the `repo`, `branch` tuple unless we solve that in another way.